ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/AbstractTaskViewModel.kt

```diff
@@ -67,10 +68,19 @@ open class AbstractTaskViewModel internal constructor(private val resources: Res
   }
 
   // TODO: Check valid values.
-  private fun validate(task: Task, value: Value?): Optional<String> =
-    if (task.isRequired && (value == null || value.isEmpty()))
-      Optional.of(resources.getString(R.string.required_task))
-    else Optional.empty()
+  private fun validate(task: Task, value: Value?): Optional<String> {
+    // Empty response for a required task.
+    if (task.isRequired && (value == null || value.isEmpty())) {
+      return Optional.of(resources.getString(R.string.required_task))
+    }
+
+    // Invalid response for draw area task.
+    if (task.type == Task.Type.DRAW_AREA && value is DrawAreaTaskIncompleteResult) {
```

Comment:
To avoid leaking subclass details into the parent here, can we instead make `validate` protected and override in `DrawAreaTaskViewModel`?

Reply:
Done